homeassistant/components/scene/powerview.py

```diff
@@ -0,0 +1,69 @@
+"""
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Gets powerview scenes from a powerview hub
+defined by a Hunter Douglas powerview app.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/scene/
+"""
+
+import logging
+
+from homeassistant.components.scene import Scene
+
+DEPENDENCIES = ['group']
+STATE = 'scening'
+_LOGGER = logging.getLogger(__name__)
+REQUIREMENTS = [
+    'https://github.com/sander76/powerviewApi/'
+    'archive/master.zip#powerview_api==0.2']
+
+HUB_ADDRESS = 'address'
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """sets up the powerview scenes stored in a powerview hub"""
+    import powerview
+
+    hub_address = config.get(HUB_ADDRESS)
+
+    _pv = powerview.PowerView(hub_address)
+    try:
+        _scenes = _pv.get_scenes()
+        _rooms = _pv.get_rooms()
+    except ConnectionError:
+        _LOGGER.exception("error connecting to powerview "
+                          "hub with ip address: %s", hub_address)
+    add_devices(PowerViewScene(hass, scene, _rooms, _pv)
+                for scene in _scenes['sceneData'])
+
+    return True
+
+
+class PowerViewScene(Scene):
+    """ A scene is a group of entities and the states we want them to be. """
+
+    def __init__(self, hass, scene_data, room_data, pv_instance):
+        self.pv_instance = pv_instance
+        self.hass = hass
+        self.scene_data = scene_data
+        self._sync_room_data(room_data)
+
+    def _sync_room_data(self, room_data):
+        room = next(room for room in room_data["roomData"]
```

Comment:
You have to pass in `None` as a second argument to `next` to have it return `None` if no iteration left. Otherwise it will raise a `StopIteration` exception.